### DISCUSSION TITLE
ManagedFileChooser issues

### DISCUSSION THREAD

**Discussion by rabbitism** [AvaloniaUI/Avalonia#11540] [Q&A] [answered]
**Describe the bug**
1. ManagedFileChooserViewModel.FileName should not be read only. It's now impossible to modify file name on saving files.
2. ManagedFileChooser samples should be fixed, it uses internal methods, developers cannot get any benefits from this demo.  

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

- OS: [e.g. Windows, Mac, Linux (State distribution)]
- Version [e.g. 0.10.0-rc1 or 0.9.12]

**Additional context**

**Comment by maxkatz6** [comment]
> ManagedFileChooserViewModel.FileName should not be read only

It's not supposed to be used by library consumers. It was even moved to Internal namespace. It is still public to support custom themes.

> ManagedFileChooser samples should be fixed, it uses internal methods

Which APIs? Normally developers use `.UseManagedSystemDialogs()` in the BuildAvaloniaApp method. That's it - all managed dialogs public API. After that StorageProvider and old SaveFileDialog APIs will use managed one internally.

**Reply by rabbitism** [reply]
For second one, yes, but the demo shows a possibility of calling managed dialogs (by constructing a new ManagedStorageProvider) in code, which is misleading.
For the first one, after using `.UseManagedSystemDialogs`, the managed save file window shows an editable file name textbox, but the value actually cannot be changed. 

![image](https://github.com/AvaloniaUI/Avalonia/assets/14807942/4d7b823b-e408-4dc1-84c5-0f5d54835bf7)
the picker result still shows predefined file name
![image](https://github.com/AvaloniaUI/Avalonia/assets/14807942/fb7766eb-bef3-4ade-859f-8d3e8c872fe3)

**Reply by rabbitism** [reply]
The demo shows a possibility of using managed file chooser on demand, but in fact it can only be used globally.

**Reply by maxkatz6** [reply, marked as answer]
> For second one, yes, but the demo shows a possibility of calling managed dialogs (by constructing a new ManagedStorageProvider) in code, which is misleading.

Only for testing, as otherwise it's not possible to have both kinds of dialogs enabled at the same time. 

> For the first one, after using .UseManagedSystemDialogs, the managed save file window shows an editable file name textbox, but the value actually cannot be changed.

This one seems to be a more specific bug that needs to be fixed, yes.
